src/scifem/solvers.py

```diff
@@ -205,15 +204,15 @@ def solve(self, atol=1e-6, rtol=1e-6, beta=1.0) -> int:
                 self._post_solve_callback(self)
 
             # Compute norm of update
-            residual = self.dx.norm(petsc4py.typing.NormType.NORM_2)
+            residual = self.dx.norm(0)
```

Comment:
1. This is not the same, norm https://petsc.org/release/manualpages/Vec/NormType/
2. Why aren't we just using: 
```suggestion
            residual = self.dx.norm(PETSc.NormType.NORM_2)
```

Reply:
Yes, this is better. My thought here was to just revert the change and go back to what we used before, but as you said it is not the same.